src/main/resources/plugin-descriptor.properties

```diff
@@ -0,0 +1,6 @@
+description=${project.description}
+version=${project.version}
+name=${project.artifactId}
+java.version=1.8
```

Comment:
is that the same as what ES requires? We don't want to cause a deployment problem by having stricter requirements than ES itself

Reply:
I have checked ES official document, these fields are required for newer ES plugins: https://www.elastic.co/guide/en/elasticsearch/plugins/5.4/plugin-authors.html

Reply:
Yes, sorry I didn't say what I meant, which was "is the minimum version of Java 1.8 a new requirement compared to base ES 5.4?". It turns out to be fine (ES 5.4 already requires 1.8), so I withdraw the original comment 😄